src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -651,26 +651,25 @@ public void Cancel()
     {
         _threadingContext.ThrowIfNotOnUIThread();
 
-        // This wait is safe.  We are not passing the async callback to DismissUIAndRollbackEditsAndEndRenameSessionAsync.
-        // So everything in that method will happen synchronously.
-        DismissUIAndRollbackEditsAndEndRenameSessionAsync(
-            RenameLogMessage.UserActionOutcome.Canceled, previewChanges: false).Wait();
+        DismissUIAndRollbackEditsAndEndRenameSession_MustBeCalledOnUIThread(
+            RenameLogMessage.UserActionOutcome.Canceled, previewChanges: false);
```

Comment:
no need for Wait. This function no longer attempts to get back to the UI thread (it requires it from callers).  And it has no async callback function to execute.  So it's fully synchronous and simple.